terraform/hcl2json.go

```diff
@@ -128,49 +108,35 @@ func parseModuleFiles(files map[string]File, vars ModuleVariables, parseRes *Par
 
 func extractModuleVariables(files map[string]File, parseRes *ParseModuleResult) ModuleVariables {
 	inputsByFile := InputVariablesByFile{}
+	localExprsMap := ExpressionMap{}
 
 	for fileName, file := range files {
-		if isValidInputVariablesFile(fileName) {
-			inputsMap, err := extractInputVariables(file)
-			if err != nil {
-				// skip non-user errors
-				if isUserError(err) {
-					parseRes.debugLogs[fileName] = GenerateDebugLogs(err)
-					parseRes.failedFiles[fileName] = err.Error()
-				}
-				continue
+		inputsMap, fileLocals, err := extractVariables(file)
```

Comment:
`extractVariables` returns input variables and local expressions for each file, so now we don't need to duplicate the error handling and for loops